### DISCUSSION TITLE
Gradle Build Failure in Mac M1

### DISCUSSION THREAD

**Discussion by ihharsh** [oppia/oppia-android#4842] [Q&A (Installation)] [answered]
Hey folks,
This is my first time getting into the open source community.
I have successfully installed the android code base in my local machine, but when i try to run the app, the gradle build fails saying
Could not resolve protoc-3.8.0-osx-aarch_64.exe (com.google.protobuf:protoc:3.8.0)

Can you please guide me regarding the same.

**Comment by adhiamboperes** [comment, marked as answer]
Hi @ihharsh, this typically happens if you skip the "Open using Rosetta" part of set up instructions to run `bash scripts/setup.sh`.

Here is how to go about it:


1. In the **Finder** app in your Mac, locate the **Applications** folder.
2. Right-click on your Terminal app and create a duplicate Terminal (and rename it accordingly, say Terminal Rosetta, to avoid confusion).
3. On the  newly created Terminal Rosetta, right-click and select "Get info", and check the option "Open using Rosetta".
4. Open Terminal Rosetta and navigate to opensource/oppia-android/.
5. Finally, run `bash scripts/setup.sh` in this Terminal and all the required files should be generated. (You should see the message that "Added protobuf platform to local properties", etc)

<img width="652" alt="Screen Shot 2023-01-13 at 11 43 33 PM" src="https://user-images.githubusercontent.com/59600948/212415765-1ab0dfb7-8229-40f2-b1ac-2ca92c22ad4b.png">

**Reply by adhiamboperes** [reply]
Hey @ihharsh do mark this as answer if it worked, or tag me incase you have any questions.

**Comment by MohitGupta121** [comment]
@ihharsh Please if your query solved please mark as answer the above comment. So that it helpful to other in the community. Otherwise we have to mark this discussion as `stale`. If you  have any query/doubt please feel free to ask in this thread.
